airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -548,3 +576,129 @@ def upload_metadata_to_gcs(bucket_name: str, metadata_file_path: Path, validator
             ],
         ),
     )
+
+
+def delete_release_candidate_from_gcs(bucket_name: str, docker_repository: str, connector_version: str) -> MetadataDeleteInfo:
+    """
+    Delete a release candidate from a GCS bucket.
+    The release candidate and version metadata file will be deleted.
+    We first check that the release candidate metadata file hash matches the version metadata file hash.
+    Args:
+        bucket_name (str): Name of the GCS bucket from which the release candidate will be deleted.
+        docker_repository (str): Name of the connector docker image.
+        connector_version (str): Version of the connector.
+    Returns:
+        MetadataDeleteInfo: Information about the files that were deleted.
+    """
+    storage_client = _get_storage_client()
+    bucket = storage_client.bucket(bucket_name)
+
+    gcp_connector_dir = f"{METADATA_FOLDER}/{docker_repository}"
+    version_path = f"{gcp_connector_dir}/{connector_version}/{METADATA_FILE_NAME}"
+    rc_path = f"{gcp_connector_dir}/{RELEASE_CANDIDATE_GCS_FOLDER_NAME}/{METADATA_FILE_NAME}"
+
+    version_blob = bucket.blob(version_path)
+    rc_blob = bucket.blob(rc_path)
+
+    if not version_blob.exists():
+        raise FileNotFoundError(f"Version metadata file {version_path} does not exist in the bucket. ")
+    if not rc_blob.exists():
+        raise FileNotFoundError(f"Release candidate metadata file {rc_path} does not exist in the bucket. ")
+    if rc_blob.md5_hash != version_blob.md5_hash:
+        raise ValueError(
+            f"Release candidate metadata file {rc_path} hash does not match the version metadata file {version_path} hash. Unsafe to delete."
+        )
```

Comment:
If we're deleting the RC from GCS, that means that we might be promoting it, in which case yes the `rc_blob.md5_hash` should equal the new default version - which is now at `version_blob.md5_hash`? OR it could mean that we're rolling back, in which case they aren't expected to be equal, right? Or am I missing something?

In the highly possible even that I'm just missing something, that tells me that maybe we should beef up the comment a bit for our future selves :).